code/modules/zombie/organs.dm

```diff
@@ -1,75 +1,73 @@
+/obj/item/organ/body_egg/zombie_infection/on_find(mob/living/finder)
 	finder << "<span class='warning'>Inside the head is a disgusting black \
 		web of pus and vicera, bound tightly around the brain like some \
 		biological harness.</span>"
 
-/obj/item/organ/zombie_infection/process()
-	if(!owner)
+/obj/item/organ/body_egg/zombie_infection/egg_process()
+	if(!ishuman(owner)) // We do not support monkey or xeno zombies. Yet.
+		qdel(src)
 		return
-	if(!(src in owner.internal_organs))
-		Remove(owner)
+	else if(reanimation_timer && (reanimation_timer < world.time))
+		zombify() // Rise and shine, Mr Romero... rise and shine.
+		reanimation_timer = null
+	else if(owner.stat == DEAD && (!reanimation_timer))
+		START_TIMER
+	else if(converts_living && !iszombie(owner) && !reanimation_timer)
+		owner << "<span class='narsiesmall'>You can feel your heart stopping, but something isn't right... \
+		life has not completely abandoned your broken form. You can only feel a deep and immutable hunger that \
+		not even death can stop! You will rise again!!!</span>"
+		START_TIMER
 
-	if(timer_id)
-		return
-	if(owner.stat != DEAD && !converts_living)
-		return
-
-	var/revive_time = rand(revive_time_min, revive_time_max)
-	var/flags = TIMER_STOPPABLE
-	timer_id = addtimer(CALLBACK(src, .proc/zombify), revive_time, flags)
-
-/obj/item/organ/zombie_infection/proc/zombify()
-	timer_id = null
+/obj/item/organ/body_egg/zombie_infection/Remove(mob/living/carbon/M, special = 0)
+	. = ..()
+	CHECK_DNA_AND_SPECIES(M)
+	if(iszombie(M) && old_species)
+		M.set_species(old_species)
 
+/obj/item/organ/body_egg/zombie_infection/proc/zombify()
+	CHECK_DNA_AND_SPECIES(owner)
 	if(!iszombie(owner))
 		old_species = owner.dna.species.type
 
-	if(!converts_living && owner.stat != DEAD)
-		return
-
-	var/stand_up = (owner.stat == DEAD) || (owner.stat == UNCONSCIOUS)
-
 	owner.grab_ghost()
 	owner.set_species(/datum/species/zombie/infectious)
+	var/old_stat = owner.stat // Save for the message
 	owner.revive(full_heal = TRUE)
-	owner.visible_message("<span class='danger'>[owner] suddenly convulses, as [owner.p_they()][stand_up ? " stagger to [owner.p_their()] feet and" : ""] gain a ravenous hunger in [owner.p_their()] eyes!</span>", "<span class='alien'>You HUNGER!</span>")
-	playsound(owner.loc, 'sound/hallucinations/far_noise.ogg', 50, 1)
-	owner.do_jitter_animation(living_transformation_time * 10)
-	owner.Stun(living_transformation_time)
-	owner << "<span class='alertalien'>You are now a zombie!</span>"
+	switch(old_stat)
+		if(DEAD, UNCONSCIOUS)
+			owner.visible_message("<span class='danger'>[owner] staggers to \
+				their feet!</span>")
+			owner << "<span class='danger'>You stagger to your feet!</span>"
+		// Conscious conversions will generally only happen for an event
+		// or for a converts_living=TRUE infection
+		if(CONSCIOUS)
+			owner.visible_message("<span class='danger'>[owner] suddenly \
+				convulses, as they gain a ravenous hunger in their \
```

Comment:
Probably worth replacing they and their with [owner.p_they()] and [owner.p_their()]. For the sake of consistency. Simiarly above, on L59.